server/lib/vision_tagger.py

```
@@ -0,0 +1,131 @@
+    "off-frame, or showing multiple garments must score below 0.6."
+)
+
+
+def _fetch_image(url: str, *, timeout: float = 15.0) -> tuple[bytes, str]:
+    """Download an image and return (bytes, mime_type)."""
+    with httpx.Client(timeout=timeout, follow_redirects=True) as client:
+        resp = client.get(url)
+        resp.raise_for_status()
+        mime = resp.headers.get("content-type", "image/jpeg").split(";", 1)[0].strip()
+        if not mime.startswith("image/"):
+            raise ValueError(f"URL did not return an image (content-type={mime!r})")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)